config/switchport.py

```diff
@@ -0,0 +1,138 @@
+
+        if "mode" in port_data:
+            existing_mode = port_data["mode"]
+        else:
+            existing_mode = "routed"
+            mode_exists_status = False
+        if (is_port and clicommon.is_port_router_interface(db.cfgdb, port)) or \
+            (not is_port and clicommon.is_pc_router_interface(db.cfgdb, port)):
+            ctx.fail("Remove IP from {} to change mode!".format(port))
+
+        if existing_mode == "routed":
+            if mode_exists_status:
+                # if the port in an interface
+                if is_port:
+                    db.cfgdb.mod_entry("PORT", port, {"mode": "{}".format(type)})
+                # if not port then is a port channel
+                elif not is_port:
+                    db.cfgdb.mod_entry("PORTCHANNEL", port, {"mode": "{}".format(type)})
+
+            if not mode_exists_status:
+                port_data["mode"] = type
+                if is_port:
+                    db.cfgdb.set_entry("PORT", port, port_data)
+                # if not port then is a port channel
+                elif not is_port:
+                    db.cfgdb.set_entry("PORTCHANNEL", port, port_data)
+
+        if existing_mode == type:
+            ctx.fail("{} is already in the {} mode".format(port,type))
+        else: 
+            if existing_mode == "access" and type == "trunk":
+                pass
+            if existing_mode == "trunk" and type == "access":
+                if clicommon.interface_is_tagged_member(db.cfgdb,port):
+                    ctx.fail("{} is in {} mode and have tagged member(s).\nRemove tagged member(s) from {} to switch to {} mode".format(port,existing_mode,port,type))
+            if is_port:
+                db.cfgdb.mod_entry("PORT", port, {"mode": "{}".format(type)})
+            # if not port then is a port channel
+            elif not is_port:
+                db.cfgdb.mod_entry("PORTCHANNEL", port, {"mode": "{}".format(type)})
+
+        click.echo("{} switched from {} to {} mode".format(port, existing_mode, type))
+
+    # if mode type is routed
+    else:
+
+        if clicommon.interface_is_tagged_member(db.cfgdb,port):
+            ctx.fail("{} has tagged member(s). \nRemove them to change mode to {}".format(port,type))
+
+        if clicommon.interface_is_untagged_member(db.cfgdb,port):
+            ctx.fail("{} has untagged member. \nRemove it to change mode to {}".format(port,type))
+
+        if "mode" in port_data:
+            existing_mode = port_data["mode"]
+        else:
+            existing_mode = "routed"
+            mode_exists_status = False
+        
+        if not mode_exists_status:
+            port_data["mode"] = type
+            if is_port:
+                db.cfgdb.set_entry("PORT", port, port_data)
+
+            # if not port then is a port channel
+            elif not is_port:
+                db.cfgdb.set_entry("PORTCHANNEL", port, port_data)
+            ctx.fail("{} is already in {} mode".format(port, type))
```

Comment:
Should this be accepted as "pass" and not need to complain that it is already in "routed" mode?  The reason I think this is better let it pass because the interface is not yet configured with any "switchport mode" and the user perhaps just want to make sure it is set to mode "routed".  And the code actually did set the mode to "routed"...

Reply:
Yes agreed, updating this one, thanks